includes/abstracts/abstract-wc-product.php

```diff
@@ -946,7 +946,13 @@ public function set_stock_quantity( $quantity ) {
 	 * @param string $status New status.
 	 */
 	public function set_stock_status( $status = '' ) {
-		$this->set_prop( 'stock_status', 'outofstock' === $status ? 'outofstock' : 'instock' );
+		$valid_statuses = wc_get_product_stock_status_options();
+
+		if ( $status && isset( $valid_statuses[ $status ] ) ) {
```

Comment:
The `$status` check is not needed here - we know it's set because it's an arg with default value.